### PR TITLE
[ISSUE #163] Add chart for operator deployment

### PR DESCRIPTION
**Pull request by drivebyer**
## What is the purpose of the change

related to https://github.com/apache/rocketmq-operator/issues/162

Add chart for operator deployment, it's best practice in k8s environment.

## Brief changelog

Add chart for operator deployment.

## Verifying this change

截屏2023-08-01 22 17 20

**Please go through this checklist to help us incorporate your contribution quickly and easily.**

Notice: `It would be helpful if you could finish the following checklist (the last one is not necessary) before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq-operator/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Check RBAC rights for Kubernetes roles.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. 
- [x] Run `make docker-build` to build docker image for operator, try your changes from Pod inside your Kubernetes cluster, **not just locally**. Also provide screenshots to show that the RocketMQ cluster is healthy after the changes. 
- [x] Before committing your changes, remember to run `make manifests` to make sure the CRD files are updated. 
- [x] Update documentation if necessary.
- [x] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: charts/rocketmq-operator/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.1.0"
```
Comment (caigy) [inline charts/rocketmq-operator/Chart.yaml:24]: The `appVersion` should be the same as that of rocketmq-operator.

**Review by caigy — CHANGES_REQUESTED** [review]
Should those definition files of CRD be placed in `templates`?

**Comment by drivebyer** [comment]
> Should those definition files of CRD be placed in `templates`?

Based on my understanding, the crds directory is used to place CRD definitions which are only installed once (https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#method-1-let-helm-do-it-for-you). Helm expects users to upgrade CRDs on their own.

**Review by caigy — CHANGES_REQUESTED** [review]
Were crd files copied into charts directory manually? If so, pls do it in `make manifest` to avoid missing that copy operation.

**Comment by drivebyer** [comment]
> Were crd files copied into charts directory manually? If so, pls do it in `make manifest` to avoid missing that copy operation.

I got you.

Actually, the files in crds directory are copied directory, but the files in templated is copy then tempalted with values.yaml.

**Comment by drivebyer** [comment]
Maybe we should remove the old deploy fashion, and migrate to helm. @caigy

**Review by caigy — APPROVED** [review]

**Review by caigy — APPROVED** [review]

**Comment by caigy** [comment]
> Maybe we should remove the old deploy fashion, and migrate to helm. @caigy

We'd better provide helm charts as another way to deploy rocketmq-operator. Although I think helm is a general way to manage app in Kubernetes, we should reserve the original way for users to choose.

**Comment by drivebyer** [comment]
> We'd better provide helm charts as another way to deploy rocketmq-operator. Although I think helm is a general way to manage app in Kubernetes, we should reserve the original way for users to choose.

Sure

**Pull request merged**